src/main/java/uk/gov/hmcts/divorce/common/event/Applicant2SolicitorApplyForFinalOrder.java

```diff
@@ -0,0 +1,239 @@
+                .map(dynamicList -> dynamicList.getValue().getLabel());
+            if (pbaNumber.isPresent()) {
+                final PbaResponse response = paymentService.processPbaPayment(
+                    updatedData,
+                    updatedDetails.getId(),
+                    updatedData.getApplicant2().getSolicitor(),
+                    pbaNumber.get(),
+                    updatedFo.getApplicant2SolFinalOrderFeeOrderSummary(),
+                    updatedFo.getApplicant2SolFinalOrderFeeAccountReference()
+                );
+
+                if (response.getHttpStatus() == CREATED) {
+                    updatedFo.updateFinalOrderWithApp2SolPaymentDetails(
+                        finalOrderFeeOrderSummary,
+                        updatedData,
+                        response.getPaymentReference(),
+                        pbaNumber.get()
+                    );
+                } else {
+                    return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                        .data(details.getData())
+                        .errors(singletonList(response.getErrorMessage()))
+                        .build();
+                }
+            } else {
+                log.error(
+                    "PBA number not present when payment method is 'Solicitor fee account (PBA)' for CaseId: {}",
+                    details.getId());
+
+                return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                    .data(updatedData)
+                    .errors(singletonList("PBA number not present when payment method is 'Solicitor fee account (PBA)'"))
+                    .build();
+            }
+        }
+
+        updatedFo.setApplicant2SolAppliedForFinalOrder(YES);
+        updatedFo.setDateApplicant2SolAppliedForFinalOrder(LocalDateTime.now(clock));
+        updatedFo.setApplicant2SolResponsibleForFinalOrder(updatedData.getApplicant2().getSolicitor().getName());
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(updatedData)
+            .state(updatedDetails.getState())
+            .build();
+    }
+
+    public SubmittedCallbackResponse submitted(CaseDetails<CaseData, State> details,
+                                               CaseDetails<CaseData, State> beforeDetails) {
+
+        log.info("{} submitted callback invoked for Case Id: {}", FINAL_ORDER_REQUESTED_APP2_SOL, details.getId());
+
+        final CaseData data = details.getData();
+        final State previousState = data.getApplication().getPreviousState();
+
+        if (AwaitingFinalOrder.equals(previousState)) {
+            log.info("Sending Applicant 2 Applied For Final Order Notification for Case Id: {}", details.getId());
+            notificationDispatcher.send(applicant2AppliedForFinalOrderNotification, details.getData(), details.getId());
+        }
+
+        if (RespondentFinalOrderRequested.equals(details.getState())) {
```

Comment:
Is it possible to hit this If statement? Will the state ever not be AwaitingFinalOrder?

Reply:
Resolved